src/styles/blocks/calendar.scss

```
@@ -0,0 +1,48 @@
+.calendar {
+  display: flex;
+  max-width: $calendar-size;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  margin: 0 auto;
+  padding: 20px 0;
+
+  @for $index from 1 through 31 {
+    &__day:nth-child(#{$index})::before {
+      content: "#{$index}";
+    }
+  }
+
+  @each $day,$dayNum in $start-day {
+    &--start-day--#{$day} .calendar__day:first-child {
+      margin-left: $dayNum * ($cell-size + $gap-size);
+    }
+  }
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} .calendar__day:nth-child(n + #{$index + 1}) {
+      display: none;
+    }
+  }
+
+  &__day {
+    position: relative;
+    @include square($cell-size);
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.5s;
```

don't use it like that, because it's the same as transition: all 0.5s
It's too hard for the browser
Check this
https://github.com/mate-academy/layout_catalog_hovers/blob/master/checklist.md

```suggestion
    transition: transform $transitionDuration;
```